services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/agent/handler/grid/GridUpdateRequestHandlerTest.java

```diff
@@ -92,236 +116,233 @@ private GridHeader buildHeader(List<Column> cols) {
+	        .thenReturn(Optional.of(header));
+
+	    // call under test
+	    GridHeader result = handler.getGridHeader(agentContext, internalConnection);
+	    assertEquals(header, result);
 	}
 
 	@Test
-	public void testHandleEventWithMissingInternalConnection() throws Exception {
-		GridUpdateRequest request = new GridUpdateRequest().setUpdate(new Update());
-		doReturn(request).when(handler).extractRequest(event);
-		when(mockGridManager.getSingletonConnection(gridSessionId, EventSource.INTERNAL)).thenReturn(Optional.empty());
-		// call under test
-		IllegalArgumentException ex = assertThrows(IllegalArgumentException.class, () -> handler.handleEvent(event));
-		assertEquals("Cannot get a grid connection.", ex.getMessage());
+	public void testGetGridHeaderWithNoHeader() {
+	    when(mockGridViewManager.readHeader(gridSessionId, internalConnection.getReplicaId(), usersReplicaId))
+	        .thenReturn(Optional.empty());
+
+	    String message = assertThrows(IllegalArgumentException.class, () -> {
+	        // call under test
+	        handler.getGridHeader(agentContext, internalConnection);
+	    }).getMessage();
+
+	    assertEquals("Cannot read the grid header.", message);
+	}
+	
+	@Test
+	public void testGetFilters() {
+	    Update update = new Update()
+	        .setFilters(List.of(
+	            new RowSelectionFilter().setIsSelected(true),
+	            new CellValueFilter().setColumnName("a").setOperator(CellValueOperator.IS_UNDEFINED)
+	        ));
+
+	    // call under test
+	    List<FilterElement> result = handler.getFilters(update);
+	    
+	    assertEquals(2, result.size());
 	}
 
 	@Test
-	public void testHandleEventWithMissingHeader() throws Exception {
-		GridUpdateRequest request = new GridUpdateRequest().setUpdate(new Update());
-		doReturn(request).when(handler).extractRequest(event);
-		GridConnectionInfo internalConn = new GridConnectionInfo().setReplicaId(11L).setSessionId(gridSessionId)
-				.setConnectionId("int-4").setSource(EventSource.INTERNAL);
-		when(mockGridManager.getSingletonConnection(gridSessionId, EventSource.INTERNAL))
-				.thenReturn(Optional.of(internalConn));
-		when(mockGridViewManager.readHeader(gridSessionId, internalConn.getReplicaId(), usersReplicaId))
-				.thenReturn(Optional.empty());
-		// call under test
-		IllegalArgumentException ex = assertThrows(IllegalArgumentException.class, () -> handler.handleEvent(event));
-		assertEquals("Grid session does not exist", ex.getMessage());
+	public void testGetFiltersWithNull() {
+	    Update update = new Update().setFilters(null);
+
+	    // call under test
+	    List<FilterElement> result = handler.getFilters(update);
+	    
+	    assertEquals(Collections.emptyList(), result);
+	}
+	
+	@Test
+	public void testCreateConValueWithJSONArray() {
+		SetValue sv = new SetValue().setColumnName("a").setValue("a string");
+		JSONObject svRaw = new JSONObject("{\"columln\":\"a\", \"value\":\"a string\"}");
```

Comment:
method under test does not look at this misspelled property

```suggestion
		JSONObject svRaw = new JSONObject("{\"column":\"a\", \"value\":\"a string\"}");
```